allo/library/nn.py

```diff
@@ -6,17 +6,24 @@
 from .systolic import systolic
 
 
-def softmax[Ty, D](X: "Ty[D, D]") -> "Ty[D, D]":
-    Z: Ty[D, D]
-    E: Ty[D, D]
-    S: Ty[D] = 0.0
-
-    for i, j in dsl.grid(D, D, name="exp_sum"):
-        E[i, j] = dsl.exp(X[i, j])
+def softmax[Ty, L](X: "Ty[L, L]") -> "Ty[L, L]":
+    Z: Ty[L, L]
+    E: Ty[L, L]
+    M: Ty[L] = -10000.0
```

Comment:
Maybe use a larger constant number, say `-0x3f3f3f3f.0` (a large enough number smaller within the range of int32)

Reply:
I could not directly use -0x3f3f3f3f.0, but I used another value which I believe is also negative enough,